programs/uxd/src/utils/calculate_credix_lp_depository_target_amount.rs

```diff
@@ -0,0 +1,41 @@
+use crate::state::controller::Controller;
+use crate::state::credix_lp_depository::CredixLpDepository;
+use crate::state::identity_depository::IdentityDepository;
+use crate::state::mercurial_vault_depository::MercurialVaultDepository;
+use crate::utils::DepositoryInfoForTargetRedeemableAmount;
+use anchor_lang::prelude::*;
+
+use super::calculate_depositories_target_redeemable_amount;
+
+pub fn calculate_credix_lp_depository_target_amount(
+    controller: &AccountLoader<Controller>,
+    identity_depository: &AccountLoader<IdentityDepository>,
+    mercurial_vault_depository: &AccountLoader<MercurialVaultDepository>,
+    credix_lp_depository: &AccountLoader<CredixLpDepository>,
+) -> Result<u64> {
+    let depositories_target_redeemable_amount = calculate_depositories_target_redeemable_amount(
+        controller.load()?.redeemable_circulating_supply,
```

Comment:
Should we call `.load()` once per depository at the start of the function? Not sure what the implications are, more computing?

Reply:
Yeah why not! applying now!